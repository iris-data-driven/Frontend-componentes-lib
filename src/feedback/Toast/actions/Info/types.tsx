export type IconType = {
  type: 'success' | 'error' | 'warning';
};

export type InfoProps = IconType & {
  title: string;
  description: string;
};
