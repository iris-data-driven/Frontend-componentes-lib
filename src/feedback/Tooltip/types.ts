export type TooltipProps = {
  id: string;
  text: string;
  type?: 'small' | 'default'
};
