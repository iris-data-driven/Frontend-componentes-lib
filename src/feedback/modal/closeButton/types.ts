export type CloseButtonType = {
  onClick?: () => void;
};
