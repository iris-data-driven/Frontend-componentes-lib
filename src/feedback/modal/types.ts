import { Styles } from 'react-modal';

type Size = 'medium' | 'large';

export type ModalProps = {
  isOpen: boolean;
  onCloseModal: () => void;
  onAfterClose?: () => void;
  onAfterOpen?: () => void;
  size?: Size;
  contentLabel?: string;
  notClosable?: boolean;
};

export type ModalStyle = (size: Size) => Styles;
