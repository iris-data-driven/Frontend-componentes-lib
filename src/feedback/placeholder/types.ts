type placeholderTypes = 'heading' | 'text' | 'circle' | 'custom';

export type PlaceholderStyleProps = {
  type?: placeholderTypes;
};

export type PlaceholderProps = PlaceholderStyleProps & {
  className?: string;
};
