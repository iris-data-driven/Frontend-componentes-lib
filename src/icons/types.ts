export type IconsProps = {
  className?: string;
  onClick?: () => void;
};
