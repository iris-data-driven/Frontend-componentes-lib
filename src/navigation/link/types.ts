export type LinkProps = {
  to: string;
  type?: 'default' | 'small';
};
