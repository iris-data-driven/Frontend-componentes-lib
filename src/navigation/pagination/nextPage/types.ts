export type NextPageProps = {
  currentPage: number;
  changePage: (page: number) => void;
  totalPages: number;
};
