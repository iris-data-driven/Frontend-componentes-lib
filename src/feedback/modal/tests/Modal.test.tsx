import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Modal from '../';

import { ModalProps } from '../types';

describe('Modal', () => {
  it('Medium modal should match snapshot', () => {
    render(<Modal isOpen={true} onCloseModal={() => {}} />);

    const component = screen.getByTestId('modal');

    expect(component).toMatchSnapshot();
  });

  it('Large modal should match snapshot', () => {
    render(<Modal size="large" isOpen={true} onCloseModal={() => {}} />);

    const component = screen.getByTestId('modal');

    expect(component).toMatchSnapshot();
  });

  it('Should render the modal component correctly', () => {
    render(<Modal isOpen={true} onCloseModal={() => {}} />);

    const component = screen.getByTestId('modal');

    expect(component).toBeInTheDocument();
  });

  it("Shouldn't render the modal component correctly", () => {
    render(<Modal isOpen={false} onCloseModal={() => {}} />);

    expect(screen.queryByTestId('modal')).not.toBeInTheDocument();
  });

  it('Should call  close modal function on click in icon ', async () => {
    const closeModal = jest.fn();
    render(<Modal isOpen={true} onCloseModal={closeModal} />);

    fireEvent.click(screen.getByRole('button', { name: /close/i }));

    expect(closeModal).toHaveBeenCalled();
  });
});
