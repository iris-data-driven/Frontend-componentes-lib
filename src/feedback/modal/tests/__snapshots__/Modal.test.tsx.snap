// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Large modal should match snapshot 1`] = `
<div
  class="sc-AxiKw jfZNqm"
  data-testid="modal"
>
  <div
    class="sc-AxgMl ehcYrB"
  />
  <div
    class="sc-AxjAm edSmYc"
  >
    <svg
      aria-label="close modal"
      class="sc-AxirZ ffwRQR"
      height="24"
      role="button"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
        fill="#F33820"
      />
      <path
        d="M8.421 8.774l7.004 7.005M15.426 8.773L8.42 15.778"
        stroke="#fff"
        stroke-linecap="round"
        stroke-width="2"
      />
    </svg>
  </div>
  <div
    class="sc-AxhCb cxUhWV"
  />
</div>
`;

exports[`Modal Medium modal should match snapshot 1`] = `
<div
  class="sc-AxiKw jfZNqm"
  data-testid="modal"
>
  <div
    class="sc-AxgMl ehcYrB"
  />
  <div
    class="sc-AxjAm edSmYc"
  >
    <svg
      aria-label="close modal"
      class="sc-AxirZ ffwRQR"
      height="24"
      role="button"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
        fill="#F33820"
      />
      <path
        d="M8.421 8.774l7.004 7.005M15.426 8.773L8.42 15.778"
        stroke="#fff"
        stroke-linecap="round"
        stroke-width="2"
      />
    </svg>
  </div>
  <div
    class="sc-AxhCb cxUhWV"
  />
</div>
`;
